pymodbus/repl/client/mclient.py

```diff
@@ -244,7 +244,7 @@ def readwrite_registers(
             read_address=read_address,
             read_count=read_count,
             write_address=write_address,
-            write_registers=write_registers,
+            values=values,
```

Comment:
I think this was also broken, as the mixin doesn't have `write_registers`.  Does anyone use `readwrite_registers`?

Reply:
Mixin should have all standard function code, independent of who uses them.

Reply:
and actually it is often used for climate controls. You read the old value  e.g. temperature and set a new one (e.g. target temperature).

I am not sure how this is broken, but if so, our tests are not correct.

Reply:
> Mixin should have all standard function code, independent of who uses them.
https://github.com/pymodbus-dev/pymodbus/blob/d2c7c778707d8e0cc1ea9ea30ffccf945ea56b89/pymodbus/client/mixin.py#L422-L428

The `write_registers` _method_ does not have a `write_registers` _argument_.  Instead it is `values`

> and actually it is often used for climate controls. You read the old value e.g. temperature and set a new one (e.g. target temperature).
> I am not sure how this is broken, but if so, our tests are not correct.

The broken functionality appears to be only in the REPL.  Users such as your mentioned climate controls don't use REPL.

The REPL tests are very limited, and don't actually check functionality.  If it was indeed broken, that's the first success for `mypy`. :)

Reply:
Very nice. The REPL server will soon be replaced by simulator, and in mid-term the REPL client will need a severe update.